packages/medusa/src/interfaces/abstract-parser.ts

```diff
@@ -0,0 +1,30 @@
+import { ParseConfig } from "papaparse"
+
+export interface IParser<TResult> {
+  parse(path: string): Promise<TResult[]>
+}
+
+export type LineContext = {
+  lineNumber: number
+}
+
+export type ParserOptions = ParseConfig
+
+export abstract class AbstractParser<TSchema, TParserResult, TOutputResult>
+  implements IParser<TParserResult>
+{
+  protected $$schema: TSchema
+
+  protected constructor(schema: TSchema) {
+    this.$$schema = schema
+  }
+
+  public abstract parse(
+    path: string,
+    options?: ParserOptions
+  ): Promise<TParserResult[]>
+
+  public abstract validateSchema(
```

Comment:
suggestion: the method purpose is more to build the output from the file content data. The schema validation is just part of the process.
We could maybe name it buildData or generateOutput or something along that line? If you have some idea 😀

Reply:
went with `buildData` 👍

Reply:
Perfect